File: ticketProject5-DistributedC/TicketSystemCC/persistance/Databases/SellerDatabase.cs
using System;
using System.Collections.Generic;
using System.Data;
using model;
using model.Validators;
using persistance.Interfaces;

namespace persistance.Database
{

    public class SellerDatabase : ISellerRepository
    {

        //private static readonly ILog log = LogManager.GetLogger("SellerDatabase");
        private Validator<Seller> validator;

        public SellerDatabase(Validator<Seller> validator)
        {
            //log.Info("Creating SellerDatabase");
            this.validator = validator;
        }

        public Seller extractEntity(List<String> attributes)
        {
            if (attributes.Count < 5)
                throw new ValidationException("Insuficient datas!\n");
            Seller seller = new Seller(attributes[1], attributes[2], attributes[3], attributes[4])
            {
                Id = Int64.Parse(attributes[0])
            };
            return seller;
        }

        protected String createEntityAsString(Seller entity)
        {
            return entity.Id + ";" + entity.FirstName + ";" + entity.LastName + ";" + entity.Username + ";" + entity.Password;
        }

        public Seller FindOne(long id)
        {
            IDbConnection con = DBUtils.getConnection();

            try
            {
                using (var comm = con.CreateCommand())
                {
                    comm.CommandText = "select id, firstName, lastName, username, password from Sellers where id=@id";
                    IDbDataParameter paramId = comm.CreateParameter();
                    paramId.ParameterName = "@id";
                    paramId.Value = id;
                    comm.Parameters.Add(paramId);

                    using (var dataR = comm.ExecuteReader())
                    {
                        if (dataR.Read())
                        {
                            long idd = dataR.GetInt32(0);
                            String firstName = dataR.GetString(1);
                            String lastName = dataR.GetString(2);
                            String username = dataR.GetString(3);
                            String password = dataR.GetString(4);

                            Seller seller = new Seller(firstName, lastName, username, password)
                            {
                                Id = idd
                            };

                            return seller;
                        }
                    }
                }
            }
            catch (Exception ex)
            {
                Console.ForegroundColor = ConsoleColor.Red;
                Console.WriteLine("Mesajul erorii este {0}", ex.Message);
            }
            return null;
        }

        public Seller FindOneByUsername(string username)
        {
            IDbConnection con = DBUtils.getConnection();

            try
            {
                using (var comm = con.CreateCommand())
                {
                    comm.CommandText = "select id, firstName, lastName, username, password from Sellers where username=@username";
                    IDbDataParameter paramUsername = comm.CreateParameter();
                    paramUsername.ParameterName = "@username";
                    paramUsername.Value = username;
                    comm.Parameters.Add(paramUsername);

                    using (var dataR = comm.ExecuteReader())
                    {
                        if (dataR.Read())
                        {
                            long idd = dataR.GetInt32(0);
                            String firstName = dataR.GetString(1);
                            String lastName = dataR.GetString(2);
                            String username1 = dataR.GetString(3);
                            String password = dataR.GetString(4);

                            Seller seller = new Seller(firstName, lastName, username1, password)
                            {
                                Id = idd
                            };

                            return seller;
                        }
                    }
                }
            }
            catch (Exception ex)
            {
                Console.ForegroundColor = ConsoleColor.Red;
                Console.WriteLine("Mesajul erorii este {0}", ex.Message);
            }
            return null;
        }

        public IEnumerable<Seller> FindAll()
        {
            IDbConnection con = DBUtils.getConnection();
            IList<Seller> sellers = new List<Seller>();

            try
            {
                using (var comm = con.CreateCommand())
                {
                    comm.CommandText = "select id, firstName, lastName, username, password from Sellers";

                    using (var dataR = comm.ExecuteReader())
                    {
                        while (dataR.Read())
                        {
                            long idd = dataR.GetInt32(0);
                            String firstName = dataR.GetString(1);
                            String lastName = dataR.GetString(2);
                            String username = dataR.GetString(3);
                            String password = dataR.GetString(4);

                            Seller seller = new Seller(firstName, lastName, username, password)
                            {
                                Id = idd
                            };

                            sellers.Add(seller);
                        }
                    }
                }
            }
            catch (Exception ex)
            {
                Console.ForegroundColor = ConsoleColor.Red;
                Console.WriteLine("Mesajul erorii este {0}", ex.Message);
            }

            return sellers;
        }


        public Seller Save(Seller entity)
        {
            if (entity == null)
                throw new ArgumentException("entity must be not null");

            validator.Validate(entity);
            //if (FindOne(entity.Id) != null)
            //{
            //    return entity;
            //}

            var con = DBUtils.getConnection();
            try
            {
                using (var comm = con.CreateCommand())
                {
                    comm.CommandText = "insert into Sellers  values (@firstName, @lastName, @username, @password)";

                    var paramFName = comm.CreateParameter();
                    paramFName.ParameterName = "@firstName";
                    paramFName.Value = entity.FirstName;
                    comm.Parameters.Add(paramFName);

                    var paramLName = comm.CreateParameter();
                    paramLName.ParameterName = "@lastName";
                    paramLName.Value = entity.LastName;
                    comm.Parameters.Add(paramLName);

                    var paramUsername = comm.CreateParameter();
                    paramUsername.ParameterName = "@username";
                    paramUsername.Value = entity.Username;
                    comm.Parameters.Add(paramUsername);

                    var paramPassword = comm.CreateParameter();
                    paramLName.ParameterName = "@password";
                    paramPassword.Value = entity.Password;
                    comm.Parameters.Add(paramPassword);

                    var result = comm.ExecuteNonQuery();
                    if (result == 0)
                        throw new Exception("Insert failed !");
                    //else
                    //log.InfoFormat("Saved the Seller {0}", entity);
                }
            }
            catch (Exception ex)
            {
                Console.ForegroundColor = ConsoleColor.Red;
                Console.WriteLine("Mesajul erorii este {0}", ex.Message);
            }

            return null;
        }


        public Seller Delete(long id)
        {
            if (id == null)
                throw new ArgumentException("Non existent user!");

            var con = DBUtils.getConnection();
            try
            {
                Seller entity = FindOne(id);

                using (var comm = con.CreateCommand())
                {
                    comm.CommandText = "delete from Sellers where id=@id";
                    IDbDataParameter paramId = comm.CreateParameter();
                    paramId.ParameterName = "@id";
                    paramId.Value = id;
                    comm.Parameters.Add(paramId);
                    var result = comm.ExecuteNonQuery();
                    if (result == 0)
                        return null;
                    else
                    {
                        //log.InfoFormat("Deleted Seller {0}", entity);
                        return entity;
                    }
                }
            }
            catch (Exception ex)
            {
                Console.ForegroundColor = ConsoleColor.Red;
                Console.WriteLine("Mesajul erorii este {0}", ex.Message);
            }
            return null;
        }

        public Seller Update(Seller entity)
        {
            if (entity == null)
                throw new ArgumentException("entity must be not null!");
            validator.Validate(entity);

            if (FindOne(entity.Id) == null)
                return entity;

            var con = DBUtils.getConnection();
            try
            {
                using (var comm = con.CreateCommand())
                {
                    comm.CommandText = "update Sellers set firstName=(@firstName), lastName=(@lastName), username=(@username), password=(@password) where id=(@id)";

                    var paramFName = comm.CreateParameter();
                    paramFName.ParameterName = "@firstName";
                    paramFName.Value = entity.FirstName;
                    comm.Parameters.Add(paramFName);

                    var paramLName = comm.CreateParameter();
                    paramLName.ParameterName = "@lastName";
                    paramLName.Value = entity.LastName;
                    comm.Parameters.Add(paramLName);

                    var paramUsername = comm.CreateParameter();
                    paramUsername.ParameterName = "@username";
                    paramUsername.Value = entity.Username;
                    comm.Parameters.Add(paramUsername);

                    var paramPassword = comm.CreateParameter();
                    paramPassword.ParameterName = "@password";
                    paramPassword.Value = entity.Password;
                    comm.Parameters.Add(paramPassword);

                    var paramId = comm.CreateParameter();
                    paramId.ParameterName = "@id";
                    paramId.Value = entity.Id;
                    comm.Parameters.Add(paramId);

                    var result = comm.ExecuteNonQuery();
                    if (result == 0)
                        return entity;
                    else
                    {
                        //log.InfoFormat("Updated Seller {0}", entity);
                        return null;
                    }
                }
            }
            catch (Exception ex)
            {
                Console.ForegroundColor = ConsoleColor.Red;
                Console.WriteLine("Mesajul erorii este {0}", ex.Message);
            }
            return entity;
        }


    }

}

File: ticketProject9-REST/TS-REST/TicketPersistance/src/main/java/ticket/persistance/jdbc/SellerDatabase.java
package ticket.persistance.jdbc;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import ticket.model.Seller;
import ticket.model.validators.ValidationException;
import ticket.model.validators.Validator;
import ticket.persistance.SellerRepository;
import ticket.persistance.jdbc.utils.JdbcUtils;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;
import java.util.Properties;


public class SellerDatabase implements SellerRepository {

    private final JdbcUtils dbUtils;
    final private Validator<Seller> validator;
    private static final Logger logger= LogManager.getLogger();

    public SellerDatabase(Validator<Seller> validator, Properties props) {
        this.validator = validator;
        dbUtils=new JdbcUtils(props);
        logger.info("Initializing SellerDatabase");
    }

    public Seller extractEntity(List<String> attributes) {
        if (attributes.size() < 5)
            throw new ValidationException("Insuficient datas!\n");
        Seller seller = new Seller(attributes.get(1), attributes.get(2), attributes.get(4));
        seller.setUsername(attributes.get(3));
        seller.setId(Long.parseLong(attributes.get(0)));
        return seller;
    }

    protected String createEntityAsString(Seller entity) {
        return entity.getId() + ";" + entity.getFirstName() + ";" + entity.getLastName() + ";" + entity.getUsername() + ";" + entity.getPassword();
    }

    @Override
    public Seller findOne(Long id) {
        if (id == null)
            throw new IllegalArgumentException("id must be not null");

        Connection con= dbUtils.getConnection();
        try {
            PreparedStatement preStmt = con.prepareStatement("select * from Sellers where id = ?");
            preStmt.setInt(1, Math.toIntExact(id));
            ResultSet result = preStmt.executeQuery();

            if (result.next()) {
                Long idd = result.getLong("id");
                String firstName = result.getString("firstName");
                String lastName = result.getString("lastName");
                String username = result.getString("username");
                String password = result.getString("password");

                Seller seller = new Seller(firstName, lastName, username, password);
                seller.setId(idd);

                result.close();
                preStmt.close();

                return seller;
            }
        } catch (Exception e) {
            System.out.println(e.toString());
        }
        return null;
    }

    @Override
    public Seller findOneByUsername(String username) {
        if (username == null)
            throw new IllegalArgumentException("username must be not null");

        Connection con= dbUtils.getConnection();
        try {
            PreparedStatement preStmt = con.prepareStatement("select * from Sellers where username = ?");
            preStmt.setString(1, username);
            ResultSet result = preStmt.executeQuery();

            if (result.next()) {
                Long idd = result.getLong("id");
                String firstName = result.getString("firstName");
                String lastName = result.getString("lastName");
                String username1 = result.getString("username");
                String password = result.getString("password");

                Seller seller = new Seller(firstName, lastName, username1, password);
                seller.setId(idd);

                result.close();
                preStmt.close();

                return seller;
            }
        } catch (Exception e) {
            System.out.println(e.toString());
        }
        return null;
    }

    @Override
    public Iterable<Seller> findAll() {
        Connection con= dbUtils.getConnection();
        List<Seller> sellers= new ArrayList<>();

        try{
            PreparedStatement preStmt = con.prepareStatement("select * from Sellers ");
            ResultSet result = preStmt.executeQuery();
            while(result.next()){
                Long id=result.getLong("id");
                String firstName = result.getString("firstName");
                String lastName = result.getString("lastName");
                String username = result.getString("username");
                String password = result.getString("password");

                Seller seller = new Seller(firstName, lastName, username, password);
                seller.setId(id);

                sellers.add(seller);
            }
            result.close();
            preStmt.close();
        }
        catch(SQLException ex){
            System.err.println("Error DB" + ex);
        }

        return sellers;
    }


    @Override
    public Seller save(Seller entity) {
        logger.traceEntry("saving Seller {}", entity);
        if (entity == null)
            throw new IllegalArgumentException("entity must be not null");
        validator.validate(entity);
//        if (findOne(entity.getId()) != null) {
//            return entity;
//        }

        Connection con= dbUtils.getConnection();
        try{
            PreparedStatement preStmt = con.prepareStatement("insert into Sellers (firstName, lastName, username, password) values (?,?,?,?)");
            preStmt.setString(1, entity.getFirstName());
            preStmt.setString(2, entity.getLastName());
            preStmt.setString(3, entity.getUsername());
            preStmt.setString(4, entity.getPassword());
            int result=preStmt.executeUpdate();

            preStmt.close();

            logger.trace("Saved {} instances", result);
            if (result == 0)
                throw new Exception("Insert Failed");
        }
        catch(SQLException ex){
            System.err.println("Error DB" + ex);
        }
        catch(Exception ex){
            logger.error(ex);
            System.err.println(ex);
        }

        logger.traceExit();
        return null;
    }

    @Override
    public Seller delete(Long id) {
        logger.traceEntry("deleting Seller with id {}", id);
        if (id == null)
            throw new IllegalArgumentException("Non existent user!");

        Connection con= dbUtils.getConnection();
        try {
            Seller entity = findOne(id);

            PreparedStatement preStmt = con.prepareStatement("DELETE FROM Sellers WHERE id = ?");
            preStmt.setInt(1, Math.toIntExact(id));
            int result = preStmt.executeUpdate();

            preStmt.close();

            //con.close();
            logger.trace("Deleted {} instances", result);
            if (result > 0)
                return entity;
            else
                return null;

        } catch (Exception e) {
            logger.error(e);
            System.out.println(e.toString());
        }

        logger.traceExit();
        return null;
    }

    @Override
    public Seller update(Seller entity) {
        logger.traceEntry("updating Seller {}", entity);
        if (entity == null)
            throw new IllegalArgumentException("entity must be not null!");
        validator.validate(entity);

        if (findOne(entity.getId()) == null)
            return entity;
        Connection con= dbUtils.getConnection();
        try {
            PreparedStatement preStmt = con.prepareStatement("update Sellers set firstName=(?), lastName=(?), username=(?), password=(?) where id=(?);");
            preStmt.setString(1, entity.getFirstName());
            preStmt.setString(2, entity.getLastName());
            preStmt.setString(3, entity.getUsername());
            preStmt.setString(4, entity.getPassword());
            preStmt.setInt(5, Math.toIntExact(entity.getId()));
            int result = preStmt.executeUpdate();

            preStmt.close();

            logger.trace("Updated {} instances", result);
            if (result > 0)
                return null;
            else
                return entity;
        } catch (Exception e) {
            logger.error(e);
            System.out.println(e.toString());
        }

        logger.traceExit();
        return entity;
    }


}
